src/nvim/os/env.c

```diff
@@ -262,8 +262,27 @@ void expand_env_esc(char_u *srcp, char_u *dst, int dstlen, bool esc, bool one,
     startstr_len = (int)STRLEN(startstr);
 
   src = skipwhite(srcp);
-  --dstlen;  // leave one char space for "\,"
+  dstlen--;  // leave one char space for "\,"
   while (*src && dstlen > 0) {
+    // Skip over `=expr`.
+    if (src[0] == '`' && src[1] == '=') {
+      size_t len;
+
+      var = src;
+      src += 2;
+      (void)skip_expr(&src);
+      if (*src == '`') {
+        src++;
+      }
+      len = (size_t)src - (size_t)var;
+      if (len > (size_t)dstlen) {
+        len = (size_t)dstlen;
+      }
+      xstrlcpy((char *)dst, (char *)var, len);
```

Comment:
If we have to save up space for the `NUL` byte after the loop, we can use `memcpy()` here.